<launch>
  <arg name="max_num_freqs" default="4" />
  <arg name="init_fft_window" default="0.25" />
  <arg name="init_fft_reeval_period" default="0.0625" />
  <arg name="sound_file_path_sub_topic" default="/smart_home/sound_file_path" />
  <arg name="sound_file_path_list_sub_topic" default="/smart_home/sound_file_path_list" />
  <arg name="fft_window_sub_topic" default="/smart_home/fft_window" />
  <arg name="playback_command_sub_topic" default="/smart_home/playback_commands" />
  <arg name="playback_frequencies_pub_topic" default="/smart_home/playback_frequencies" />
  <arg name="playback_updates_pub_topic" default="/smart_home/playback_updates" />

  <node
    name="SfAnalyzer"
    pkg="smart_home_sfp"
    type="SfAnalyzer"
    output="screen"
    required="true"
    args="$(arg max_num_freqs)
          $(arg init_fft_window)
          $(arg init_fft_reeval_period)
          $(arg sound_file_path_sub_topic)
          $(arg sound_file_path_list_sub_topic)
          $(arg fft_window_sub_topic)
          $(arg playback_command_sub_topic)
          $(arg playback_frequencies_pub_topic)
          $(arg playback_updates_pub_topic)"/>
</launch>
